app/views/courses/CoursesView.coffee

```diff
@@ -81,6 +85,13 @@ module.exports = class CoursesView extends RootView
     if hocCourseInstance
       @courseInstances.remove(hocCourseInstance)
 
+    _.forEach Object.keys(courseIDs), (courseID) =>
+      levels = new Levels()
+      @listenTo levels, 'sync', =>
+        @originalSlugMap[level.get('original')] = level.get('slug') for level in levels.models
+        @render?()
```

Comment:
The `?` should only be needed if the view is destroyed, but if that's the case the previous line will have broken since `@originalSlugMap` would also be gone. Maybe just check `if @destroyed` at the beginning?